text/0090-tikv-gc.md

```diff
@@ -0,0 +1,88 @@
+# RFC: TiKV RawKV MVCC GC
+
+
+## Summary
+Move TiKV MVCC GC worker from TiDB into a group of independent GC worker node role and implement a new GC process in TiKV for RawKV.
+
+## Motivation
+1.GC worker is an important component for TiKV that deletes outdated MVCC data so as to not explode the storage. But currently, the GC worker is implemented in TiDB, which makes TiKV not usable without TiDB.And current GC process is just for transaction of TiDB,it's not usable for RawKV.  
+2.Standardize the API used to set and obtain GC status in PD to improve the developer's experience.
+
+## Background
+According to the documentation for the current GC worker in a TiDB cluster, the GC process is as follows:
+
+In TiDB GC worker leader:
+1. Regularly calculates a new timestamp called "GC safe point"(The default interval is 10min), and push the safe point to PD.
+2. Get the minimal Service safe point  among all services from the response of step 2, which is GC safe point .
+3. Txn GC process: resolve locks and record delete ranges information.
+
+In PD leader:
+1. Receive update safe point requests from TiDB or other tools (e.g. CDC, BR).
+2. Calculate the minimal timestamp = min(all service safe point, now - gc_life_time).
+
+the GC safe point data in etcd of PD as follows:  
+- safe point generated by TiDB:
+     ```shell
+  /gc/safe_point/service  
+    ```
+- service safe point generated by CDC,BR or Lighting:  
+     ```shell
+  /gc/safe_point/service/$serviceID
+    ```
+
+In every TiKV nodes：
+1. Get GC safe point from PD regularly.
+2. Deletion will be triggered in CompactionFilter and GcTask thread;
+   
+## New GC worker architecture
+In a TiKV cluster without TiDB nodes , there are a few different points as follows:
+1. We need to move GC worker into another node role.
+2. For [API V2](https://github.com/tikv/rfcs/blob/master/text/0069-api-v2.md) .It need gc the earlier version in default cf. But Txn GC worker process will be triggered by WriteCompactionFilter of write cf.
+3. RawKV encoded code is different with Txn data in TiDB.
+
+So we designed a new GC architecture and process for TiKV cluster.It will be extended on the original interface to support the RawKV MVCC GC. For the original TiDB scenario, the old GC implementation can be used first.
+
+## Detailed design
+For support RawKV GC in TiKV cluster deploy without TiDB nodes.
+1. Add a new node role instead of GC worker in TiDB nodes.
+- Why we choose to create a new node role:
+  - IF we add GC Worker in PD: It will cause the problem of client-go circular dependency.
```

Comment:
I think the main reason is that PD is designed for scheduling & meta storage. Acting as GC worker will bring inappropriate feature and lower availability of PD.